### DISCUSSION TITLE
How i can export aab in adobe aniamte ??

### DISCUSSION THREAD

**Discussion by avcr-wq** [airsdk/Adobe-Runtime-Support#1027] [Q&A]
Hello
I wish good and happiness to everyone. I program my own applications through Adobe Animate 2021. Recently, Google changed the application's file upload policy and required programmers to upload files in aab format. Unfortunately, the Adobe Animate program does not export files except apk, so how do I solve the problem, especially since my applications are periodically updated on the store
How i can export aab in adobe animate ??

**Comment by ajwfrost** [comment]
There's a post created below that explains some of it. If you're using Animate that doesn't (yet?) have a UI option to select the App Bundle format, then the best thing to do is edit the "adt.cfg" file in the latest AIR SDK release and enable the line:
```
CreateAndroidAppBundle=true
```
Then if you run the normal packaging through Animate, you should end up with a file that claims to be an APK file - but actually it will be an AAB file.

Note - please delete any existing file with the output name before you start otherwise it gets corrupted.

https://www.reddit.com/r/as3/comments/lm6cvj/how_to_create_aab_bundle_with_harman_air/

thanks

**Comment by Jason0247** [comment]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/91530077/c28461e6-c603-42d4-8265-e6cd9bb4c70d)
After modifying the configuration file and deleting the APK in the original directory for republishing, an error will still be reported.AIR SDK5.2.2.5
@ajwfrost

**Reply by marchbold** [reply]
He means that the actual apk file that is output from animate is actually an aab, simply rename it to aab and you'll have your aab file to publish. This is because animate specifies the output of adt to be a specific output filename `app.apk` (with the apk extension), and the `CreateAndroidAppBundle` overrides the target (apk -> aab) but not the filename.

**Reply by Jason0247** [reply]
I don't know why. After changing the configuration file, I couldn't publish it and reported an error. Have you encountered this problem? thanks

**Reply by marchbold** [reply]
What was the error ? AAB will likely require that you have Android SDK installed (easiest done by installing android studio). And have you set the path to the android sdk in your config?

**Reply by Jason0247** [reply]
Android Gradle-based package creation needs an Android SDKpath to be provided [-platformsdk]
Thank you, then I should understand. The reason for the error may be that I have not installed the Android SDK. Aab is only used when uploading stores, right? If you don't upload the store, will APK not allow installation?

**Reply by marchbold** [reply]
AAB is a package of multiple architectures and yeah generally only used to upload to the stores. You can use bundletool to create an apk and install it if you need to test from the AAB directly.

**Reply by Jason0247** [reply]
I understand, thank you. If that's the case, we can study it later if necessary.

**Reply by ajwfrost** [reply]
On other thing to add: the publishing of an APK now should also be done using Gradle. So it's generally better to have Android Studio installed and with an Android SDK set up, because then the APK will be similar to the AAB when you create that. The older mechanism that we had for creating APKs without Gradle is often failing now because of the Android tools that have moved on a bit (particularly resource compilation).

FYI the `adt -installApp` command can be used on both an APK file and an AAB file. (If you ask to install an AAB file then we just use the bundletool utility). But yes, it's slightly quicker and uses less space if you just use APKs during development..

thanks
